### PR TITLE
use yarn

### LINKED ISSUE #9: automate the install of all dependencies in the modules in src
Currently we are using a hack in CI to make test pass.

A "proper", cross-platform fix is needed so `npm test` works on windows too.

See #8 for more details.

### PR DESCRIPTION
**Pull request by simoneb**
fixes #9

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,4 +2,5 @@ node_modules/
 .nyc_output/
 .DS_Store
 package-lock.json
+yarn.lock
```
Comment (mcollina) [inline .gitignore:5]: yarn really needs the lockfile unfortunately :(. I'm also ok in ahving the lockfile for trainings.

**Review by mcollina — CHANGES_REQUESTED** [review]
I think you need to rebase on top of master

**Comment by simoneb** [comment]
> I think you need to rebase on top of master

Yes I did this change on top of an older commit just to show what it would look like

File: .gitignore (hunk 1 above)
Comment (simoneb) [reply .gitignore:5]: Why do you think it needs it?

**Comment by simoneb** [comment]
closing as this change has been pushed to master

**Comment by simoneb** [comment]
@mcollina I pushed to master which implement this change by:

- reverting your last 2 commits
- using yarn everywhere

by using yarn workspaces:

- we can install packages in the root only (as it was before your change)
- we can avoid using npx when running tests in nested packages because yarn will lookup the tap bin in the monorepo root

it's a win-win

**Pull request closed without merging**